core/src/main/java/com/graphhopper/routing/AbstractRoutingAlgorithm.java

```diff
@@ -56,8 +57,9 @@ public AbstractRoutingAlgorithm(Graph graph, Weighting weighting, TraversalMode
         this.traversalMode = traversalMode;
         this.graph = graph;
         this.nodeAccess = graph.getNodeAccess();
-        outEdgeExplorer = graph.createEdgeExplorer(DefaultEdgeFilter.outEdges(flagEncoder));
-        inEdgeExplorer = graph.createEdgeExplorer(DefaultEdgeFilter.inEdges(flagEncoder));
+        inEdgeFilter = DefaultEdgeFilter.inEdges(flagEncoder.getAccessEnc());
+        outEdgeFilter = DefaultEdgeFilter.outEdges(flagEncoder.getAccessEnc());
+        edgeExplorer = graph.createEdgeExplorer();
     }
```

Comment:
I already removed the in/outEdgeExplorer for #1835 here. I left a few todos where we can simply move the access flags into the weighting for #1835.